api/logs/views.py

```diff
@@ -29,8 +32,24 @@ def get_log(self):
             )
         return log
 
+    def check_log_permission(self, log):
+        """
+        Cycles through nodes on log backrefs.  If user can view any of the nodes, the log can be viewed.
+        """
+        auth_user = get_user_auth(self.request)
+        log_nodes = []
+
+        for node_id in log._backrefs['logged']['node']['logs']:
+            node = get_object_or_error(Node, node_id, display_name='node')
+            log_nodes.append(node)
+            if node.can_view(auth_user):
+                return True
+
+        self.check_object_permissions(self.request, log_nodes[0])
+        return False
```

Comment:
Since the return value isn't used, how about:

 - dropping this `return False`
 - a plain `return` just above, instead of `return True`
 - adding the content at https://github.com/CenterForOpenScience/osf.io/pull/4996/files#r52082314 to the docstring or in comments